### PR TITLE
web: visualize changes on files using treemaps

### PR DESCRIPTION
**Pull request by flepied**
- new requests: changes_by_file_map and authors_by_file_map
- files param (regexp to filter on changed_files.path)

### CODE REVIEW

File: monocle/db/queries.py [hunk 1]
```diff
@@ -759,3 +764,44 @@ def new_contributors(es, index, repository_fullname, params):
     )
     diff = new.difference(old)
     return {'items': [n for n in new_authors if n['key'] in diff]}
+
+
+def changes_by_file_map(es, index, repository_fullname, params):
```
Comment (morucci) [inline monocle/db/queries.py:769]: Could you add the support for size. And also order the returned values.

File: monocle/db/queries.py [hunk 2]
```diff
@@ -759,3 +764,44 @@ def new_contributors(es, index, repository_fullname, params):
...
+        for f in change['changed_files']:
+            key = '{}:{}'.format(change['repository_fullname'], f['path'])
+            try:
+                files[key] += 1
+            except KeyError:
+                files[key] = 1
+    return {'changes': files}
+
+
+def authors_by_file_map(es, index, repository_fullname, params):
```
Comment (morucci) [inline monocle/db/queries.py:789]: Do you think we could get the change rate by file for each authors in the returned values ?

File: monocle/db/queries.py (hunk 1 above)
Comment (flepied) [reply monocle/db/queries.py:769]: the values are in a dict so there is no need for sorting.

File: monocle/db/queries.py (hunk 2 above)
Comment (flepied) [reply monocle/db/queries.py:786]: we'll have to define what it really means but I'd prefer to that in another PR.

**Comment by morucci** [comment]
I think that's a nice to have this treemap. I was able to navigate and saw my changes in zuul/.* The tooltip is hard to understand "1MB, 7/9" <- What is this ?

**Comment by flepied** [comment]
> I think that's a nice to have this treemap. I was able to navigate and saw my changes in zuul/.* The tooltip is hard to understand "1MB, 7/9" <- What is this ?

yes I tried to disable it but I think it is computing the sum of all the numbers below.

**Pull request merged**